src/utils/verifiable-credential/snapVC.ts

```diff
@@ -63,31 +63,35 @@ export class SnapVerifiableCredential extends BaseVerifiableCredential {
     ],
   };
 
-  protected getSubjectDid(snapId: string): SnapDid {
-    return `snap://${snapId}`;
+  protected getSubjectDid(snapChecksum: string): SnapDid {
+    return `snap://${snapChecksum}`;
```

Comment:
Snap checksum can contain `/`, which makes `snap://${snapChecksum};` an invalid URL anymore, I think this will bring problems in the future.

Suggestion is to convert to [base64url](https://en.wikipedia.org/wiki/Base64#Variants_summary_table) and use Opaque types same as [`SemverVersion` and `SemverRange` to differentiate](https://github.com/MetaMask/utils/blob/e97f2460d23ec9ed0907e5b68af8ecb84b28cb5a/src/versions.ts#L33-L56) between base64 and base64url.